src/server/internalServer.cpp

```diff
@@ -772,7 +774,13 @@ std::unique_ptr<Response> InternalServer::handle_content(const RequestContext& r
 
   const std::shared_ptr<Reader> reader = get_reader(bookName);
   if (reader == nullptr) {
-    return Response::build_404(*this, request, bookName);
+    MustacheData data;
+    data.set("root", m_root);
+    data.set("pattern", pattern);
+    kainjow::mustache::mustache tmpl("Make a full text search for <a href=\"{{{root}}}/search?pattern={{{pattern}}}\">{{pattern}}</a>");
```

Comment:
At this point `pattern` is URL-encoded (since it was extracted as a fragment of a URL). When generating the HTML, pattern must stay URL-encoded in the search URL, but must be URL-decoded as text to be presented to the user (and we let mustache HTML-encode it).

The best solution may be to compose the search URL via C++ and pass it as a single parameter to mustache. Then you can create and share a mustache template with the other fragment of related code.